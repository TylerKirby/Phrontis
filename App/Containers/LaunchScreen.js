import React from 'react'
import { Text, View, Alert } from 'react-native'
import RoundedButton from '../Components/RoundedButton'

// Styles
import styles from './Styles/LaunchScreenStyles'

export default class LaunchScreen extends React.Component {
  static navigationOptions = {
    header: null
  }

  componentDidMount() {
    Alert.alert(
      'Alert Title',
      'My Alert Msg',
      [
        {text: 'Ask me later', onPress: () => console.log('Ask me later pressed')},
        {text: 'Cancel', onPress: () => console.log('Cancel Pressed'), style: 'cancel'},
        {text: 'OK', onPress: () => console.log('OK Pressed')},
      ],
      { cancelable: false }
    )
  }
  render () {
    return (
      <View style={styles.container} >
        <Text style={styles.title}>
          Phrontis
        </Text>
        <Text style={styles.subtitle}>
          An Ancient Greek Study Companion App
        </Text>
        <View>
          <RoundedButton
            text='BEGIN'
            onPress={() => this.props.navigation.navigate('UnitSelectScreen')}
          />
        </View>
      </View>
    )
  }
}
