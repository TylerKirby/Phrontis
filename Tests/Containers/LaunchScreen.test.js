import 'react-native';
import React from 'react';
import LaunchScreen from '../../App/Containers/LaunchScreen';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

test('LaunchScreen component renders correctly', () => {
  const tree = renderer.create(<LaunchScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
