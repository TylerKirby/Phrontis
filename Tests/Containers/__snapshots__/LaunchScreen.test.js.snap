// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchScreen component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0496FF",
      "flex": 1,
      "justifyContent": "center",
      "paddingBottom": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#FFBC42",
        "fontFamily": "ArialHebrew-Light",
        "fontSize": 60,
        "textAlign": "center",
      }
    }
  >
    Phrontis
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#FFBC42",
        "fontFamily": "ArialHebrew-Light",
        "fontSize": 20,
        "textAlign": "center",
      }
    }
  >
    An Ancient Greek Study Companion App
  </Text>
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#D81159",
          "borderRadius": 5,
          "height": 45,
          "justifyContent": "center",
          "marginHorizontal": 25,
          "marginVertical": 10,
          "opacity": 1,
          "width": 150,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginVertical": 10,
            "textAlign": "center",
          }
        }
      >
        BEGIN
      </Text>
    </View>
  </View>
</View>
`;
